### DISCUSSION TITLE
Proper way to close a dialog when clicking outside of it ..

### DISCUSSION THREAD

**Discussion by berndmehnert** [AvaloniaUI/Avalonia#7903] [Q&A]
Hi, 
I wonder what the recommended way is to close a dialog by clicking outside of it. Two things should happen when I click outside:
1) The dialog closes (here one could use f.e. the Window.Deactivated event and write an appropriate handler ..)
2) The click outside the dialog-window possibly effects controls we clicked on unintentionally, f.e. buttons, so we might want to prevent this to happen. The handler in 1) could somehow deal with that. To solve 2) may be one should deactivate all other windows, if this is possible?
My question would be: What is the proper way to handle the two points, especially 2)? 
Thank you very much for any comment ;) ..

**Comment by maxkatz6** [comment]
> may be one should deactivate all other windows

Not sure if I understand the problem. But it seems you need ```window.Activate()``` method. It supposed to deactivate other windows.

**Comment by timunie** [comment]
Tbh I use dialogs provided by other libs, for example FluentAvalonia or Aura.UI. they render a nice overlay and you can't click any other button. Maybe give it a try. I think you can also define what to do if the user clicks the overlay, but I don't know how hard that would be.

Happy coding
Tim
